sdcm/scan_operation_thread.py

```diff
@@ -0,0 +1,352 @@
+                    if limit and limit < ck_range:
+                        # Example: select * from scylla_bench.test where pk = 1 and ck > 10 and ck < 15
+                        # order by ck desc limit 3
+                        # output of ck should be: [12,13,14]
+                        normal_query += \
+                            f' and {ck_name} < {ck_random_max_value} and {ck_name} >= {ck_random_max_value - limit}'
+                    else:
+                        normal_query = reversed_query
+                else:
+                    reversed_query += self.reversed_query_filter_ck_by[ck_filter].format(ck_name,
+                                                                                         ck_random_min_value)
+
+                    # example: rows-count = 20, ck > 10, limit = 5 ==> ck_range = 20 - 10 = 10 ==> limit < ck_range
+                    # reversed query is: select * from scylla_bench.test where pk = 1 and ck > 10
+                    # order by ck desc limit 5
+                    # normal query should be: select * from scylla_bench.test where pk = 1 and ck > 15 limit 5
+                    if ck_filter == 'gt':
+                        ck_range = rows_count - ck_random_min_value
+                        if limit and limit < ck_range:
+                            normal_query += self.reversed_query_filter_ck_by[ck_filter].format(ck_name,
+                                                                                               rows_count - limit)
+                        else:
+                            normal_query = reversed_query
+
+                    # example: rows-count = 20, ck < 10, limit = 5 ==> limit < ck_random_min_value (ck_range)
+                    # reversed query is: select * from scylla_bench.test where pk = 1 and ck < 10
+                    # order by ck desc limit 5
+                    # normal query should be: select * from scylla_bench.test where pk = 1 and ck >= 5 limit 5
+                    if ck_filter == 'lt':
+                        if limit and limit < ck_random_min_value:
+                            normal_query += f' and {ck_name} >= {ck_random_min_value - limit}'
+                        else:
+                            normal_query = reversed_query
+                reversed_query += f' order by {ck_name} {self.reversed_order}'
+
+                normal_query += query_suffix
+                reversed_query += query_suffix
+                self.log.info('Randomly formed normal query is: %s', normal_query)
+                self.log.info('[scan: %s, type: %s] Randomly formed reversed query is: %s', self.scans_counter,
+                              ck_filter, reversed_query)
+            else:
+                self.log.info('No partition keys found for table: %s! A reversed query cannot be executed!', self.ks_cf)
+                return None
+        return normal_query, self.randomly_add_timeout(reversed_query)
```

Comment:
Let's remove the timeout from these queries.

Reply:
removed.